shoop/campaigns/modules.py

```diff
@@ -36,13 +36,13 @@ def discount_price(self, context, product, price_info):
         for campaign in CatalogCampaign.get_matching(context, shop_product):
             price = price_info.price
             # get first matching effect
-            effect = campaign.effects.first()
-            price -= effect.apply_for_product(context=context, product=product, price_info=price_info)
-            if best_discount is None:
-                best_discount = price
+            for effect in campaign.product_effects.all():
+                price -= effect.apply_for_product(context=context, product=product, price_info=price_info)
+                if best_discount is None:
+                    best_discount = price
```

Comment:
Neither this `if best_discount is None` shouldn't be in the loop. Don't indent it.